rules/the-best-dependency-injection-container/rule.md

```diff
@@ -44,17 +42,69 @@ The top tools all contain comparable functionality. In practice which one you us
 **Important:** Unless a specific shortfall is discovered with the container your team uses, you should continue to use the same container across all of your projects, become an expert with it and invest time on building features rather than learning new container implementations.
 
 ::: bad  
-![Figure: Bad Example - Ninject was a top container but is no longer developed as actively as Autofac and Structuremap. Both Autofac and Structuremap have active communities and contributors that ensure they stay up to date with the latest changes in .Net](dic-bad.png)  
+![Figure: Bad Example - Ninject and StructureMap were top containers but are no longer actively developed. Together with Autofac, they do not support the latest version of .NET](di-container-bad.png)  
+:::
+
+```csharp
+using SSW.SugarLearning.DependencyResolver;
+using StructureMap;
+
+namespace SSW.SugarLearning.WebJob.BadgeTask
+{
+    public class Program
+    {
+        private const string ConfigKey_AppInsightInstrumentationKey = "AppInsightInstrumentationKey";
+
+        private static void Main()
+        {
+            var environment = ConfigHelper.GetEnvironmentLabel();
+            var notifyUserOnDisqualification = environment == EnvironmentLabel.Production;
+            IContainer container = IoC.Initialize(); 
+            new BadgeTaskJob(Program.ConfigKey_AppInsightInstrumentationKey, container, notifyUserOnDisqualification).Run();
+        }
+    }
+}
+```
+::: bad
+Bad example - Use the StructureMap IoC container but did not do the proper dependency injection
 :::
 
-::: good  
-![Figure: Good Example - Autofac has a great combination of performance and features and is actively developed](dic-good.png)  
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddApplication();
```

Comment:
Can we have a simpler example here?  There's lots of code that's not relevant to IoC.

Reply:
Removed irrelevant code